cmd/agent/install_script.sh

```diff
@@ -586,8 +586,8 @@ restart_cmd="$sudo_cmd $service_cmd datadog-agent restart"
 stop_instructions="$sudo_cmd $service_cmd datadog-agent stop"
 start_instructions="$sudo_cmd $service_cmd datadog-agent start"
 
-if command -v systemctl 2>&1; then
-  # Use systemd if systemctl binary exists
+if [[ `ps --no-headers -o comm 1 2>&1` == "systemd" ]] && command -v systemctl 2>&1; then
```

Comment:
Are there cases where `ps` may not be able to see process `1` (eg. if the `hidepid` option is set to `1` or `2` on `/proc`)? Should we add `$sudo_cmd` to this command?

Reply:
I didn't know about `hidepid`. Running it as root should fix that, I'll update the command.